packages/dev/core/src/Engines/Extensions/engine.readTexture.ts

```diff
@@ -167,16 +170,35 @@ ThinEngine.prototype._readTexturePixelsSync = function (
     }
 
     if (flushRenderer) {
-        this.flushFramebuffer();
+        engine.flushFramebuffer();
     }
 
-    gl.readPixels(x, y, width, height, gl.RGBA, readType, <DataView>buffer);
+    return { outputBuffer: buffer, readType };

```

Comment:
can we avoid creating an object here for better GC

Reply:
Sure.